.github/workflows/10-linters-tests.yaml

```diff
@@ -0,0 +1,99 @@
+# This workflow run linters and tests.
+# It is supposed to be run on each push to main branch, as well as for each push in pull request.
+
+name: Linters
+
+on:
+  pull_request:
+    branches: [ '*' ]
+
+env:
+  GO_VERSION: "1.21"
+  IMAGE: linode/linode-cosi-driver
+
+permissions:
+  contents: read
+
+jobs:
+  commitlint:
+    name: Lint commit messages
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: wagoid/commitlint-github-action@v5
+
+  golangci-lint:
+    name: Run golangci-lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: ${{ env.GO_VERSION }}
+          cache: false
+      - uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
+          skip-cache: true
+
+  shell-linter:
+    name: Run Shellcheck, Checkmake and Perl Critic
```

Comment:
Perl Critic?

Reply:
Whoopsie, gonna clean this name, I copied it from my template 😮‍💨